src/gevent/_hub_primitives.py

```diff
@@ -168,12 +168,23 @@ def _cleanup(self):
                 except: # pylint:disable=bare-except
                     traceback.print_exc()
 
+    def __enter__(self):
+        return self
+
+    def __exit__(self, typ, value, tb):
+        self._cleanup()
+
 
 def iwait_on_objects(objects, timeout=None, count=None):
     """
     Iteratively yield *objects* as they are ready, until all (or *count*) are ready
     or *timeout* expired.
 
+    This function allocates resources which must be cleaned up. Consuming the
```

Comment:
Could we please get a `.. versionchanged` directive added at the end?

Reply:
Done!